src/App.js

```diff
@@ -1,16 +1,45 @@
 import React from 'react';
 import './App.css';
-import chatMessages from './data/messages.json';
+import entries from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
+
 
 const App = () => {
+  const [messages, setMessages] = useState(entries);
+
+  const handleLikeToggle = (id) => {
+    setMessages((prevMessages) => {
+      const updatedMessages = prevMessages.map((message) => {
+        if (id === message.id) {
+          return {
+            sender: message.sender,
+            body: message.body,
+            timeStamp: message.timeStamp,
+            liked: !message.liked,
+            id: message.id,
+          };
+        } else {
+          return message;
+        }
+      });
+      return updatedMessages;
+    });
+  };
+
+  const totalLikedCount = messages.filter((message) => message.liked).length;
```

Comment:
Nice, I've seen some people use `reduce` for this but taking the length of filtering on liked messages is a cool approach too!